# HG changeset patch
# User Rik <rik@octave.org>
# Date 1421606291 28800
#      Sun Jan 18 10:38:11 2015 -0800
# Node ID 40b0f41a710088591bb7cd9a11d7be597d5cb751
# Parent  e33427a854ec8bb6269231e84a986daca82a8737
Print black&white when using -dps device with gnuplot (bug #43975).

* __go_draw_axes__.m: Check the value of mono before adding cmap to
plot stream.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -1626,17 +1626,17 @@ function __go_draw_axes__ (h, plot_strea
              __do_enhanced_option__ (enhanced, hlgnd));
   else
     fputs (plot_stream, "unset key;\n");
   endif
   fputs (plot_stream, "set style data lines;\n");
 
   cmap = [cmap; addedcmap];
   cmap_sz = cmap_sz + rows (addedcmap);
-  if (length (cmap) > 0)
+  if (mono == false && length (cmap) > 0)
     fprintf (plot_stream,
              "set palette positive color model RGB maxcolors %i;\n",
              cmap_sz);
     fprintf (plot_stream,
              "set palette file \"-\" binary record=%d using 1:2:3:4;\n",
              cmap_sz);
     fwrite (plot_stream, [1:cmap_sz; cmap.'], "float32");
     fwrite (plot_stream, "\n");
